machine.go

```diff
@@ -405,45 +449,70 @@ func (m *Machine) writerKernelModules(w *writerhelper.WriterHelper, moddir strin
 			"modules.symbols",
 			"modules.symbols.bin",
 			"modules.builtin.bin",
-			"modules.devname")
+			"modules.devname"}
 
-	// build a list of built-in modules so that we don’t attempt to copy them
-	var builtinModules = make(map[string]bool)
 
-	f, err := os.Open(path.Join(moddir, "modules.builtin"))
-	if err != nil {
-		return err
+	for _, v := range modfiles {
+		if err := w.CopyFile(moddir + "/" + v); err != nil {
+			return err
+		}
 	}
-	defer f.Close()
 
-	scanner := bufio.NewScanner(f)
-	for scanner.Scan() {
-		module := scanner.Text()
-		builtinModules[module] = true
+	prefix := ""
+	if mergedUsrSystem() {
+		prefix = "/usr"
 	}
 
-	if err := scanner.Err(); err != nil {
-		return err
-	}
+	copiedModules := make(map[string]bool)
 
 	for _, v := range modules {
-		if builtinModules[v] {
+		modname := strings.TrimSuffix(v[strings.LastIndex(v, "/") + 1 :], ".ko")
```

Comment:
Instead of this convoluted massage, simply store the module without the path/extension.

Reply:
will update accordingly